src/models/auth.js

```diff
@@ -12,9 +14,76 @@ export default {
         payload,
       });
     },
-    *logoutUser({ payload }, { put }) {
+    /* eslint-disable camelcase */
+    // TODO: Design a basic codeMap
+    // for general purpose messages.
+    *loginUser({ payload }, { call }) {
+      try {
+        const response = yield call(queryLoginUser, payload);
+        const { message, auth_token } = response;
+        return {
+          message,
+          auth_token,
+          status: 'success',
+        };
+      } catch (error) {
+        const { data } = error;
```

Comment:
@gurbirkalsi I think that we haven't been seriously thinking about the wording and form of error messages yet, but we definitely should since REST clients deserve messages as good as dashboard users. So, yes; when the server returns a failure HTTP status we should always return an error document with a reliable and meaningful `"message"` string that you should be able to present directly. As I mentioned earlier to Aquib, we could also consider producing an "errorcode" token that would uniquely identify a message regardless of arguments or potential future localization in case you need to distinguish errors at a finer granularity than HTTP status.

If we get to the point where we want to produce messages that might benefit from HTML formatting, we can discuss at that time how we want to handle it.

Reply:
I definitely support having the server return an "errorcode" token; however, I would recommend that the conversion of that to a human-readable form be done as close to the human as feasible.  This enables custom formatting, internationalization, and basic de-coupling.  I would counsel against putting this burden on the server -- the presentation of errors belongs in the UI.

Now, you bring up an interesting point with REST clients, but the audience for diagnostics there is different.  For one thing, they can be referred to documentation for translating messages.  Also, they don't have expectations for formatting, and it's reasonable to offer no internationalization in that context.  Mostly, I think we want to avoid creating a temptation for the client to simply pass server message text on to a human user.

Reply:
We used a `codeMap` https://github.com/distributed-system-analysis/pbench-dashboard/blob/6e259c44e9f46de7974a73ff3667180594015117/src/utils/request.js#L5. Maybe we can do something similar.

Reply:
Client-side message localization requires continued close coordination, especially when there are parameters involved; the server will need to send the raw parameters along with the catalog ID and the "prototype" unlocalized message. There's no reason we can't do that, but it does present additional costs. It does also simplify browser-specific formatting in the messages that we wouldn't want the server to do, so there can be benefits as well. If we're going to go that way, we should target a server error document something like:

```
{
    "message_code": "LOGIN_ERROR",
    "message": "Unable to login user Abominable",
    "parameters": {
        "username": "Abominable"
    }
}
````

(In any case, we should be considering moving to a message catalog within the server, if for no other reason that message grammar and content can be easily reviewed separately, e.g. by documentation, without going through the code to find them all.)